syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -87,14 +92,13 @@ def retrieve_shares(self):
         pointers = self.retrieve_pointers()
         shares = []
         for pointer in pointers:
-            shares.append(pointer.copy().get())
+            shares.append(pointer[0].copy().get())
         return shares
 
     def retrieve_pointers(self):
         shares_map = self.__get_shares_map()
         players = list(shares_map.keys())
-        pointers = list(shares_map[players[0]])
-        pointers.append(shares_map[players[1]][1])
+        pointers = [shares_map[player] for player in players]
```

Comment:
Maybe you can simply put here: `pointers = shares_map.values()` ?

Reply:
In python3.8 (I am not sure if this is also in python3.7):
```
>>> a = {'a': 3, 'b': 4}
>>> a
{'a': 3, 'b': 4}
>>> a.values()
dict_values([3, 4]) <--- those are dict values
```

Reply:
Ok, maybe `list(shares_map.values())`  to be sure?